File: src/RemoveDuplicatesFromSortedArray.java
import java.util.Arrays;

/**
 * User: guoxc
 * Date: 15-1-13
 * Time: 15:42
 */
public class RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] A){
        if(A.length == 0) return 0;
        int i = 0;
        int j = 0;
        while(i < A.length) {
            while (i < A.length && A[i] == A[j]) {
                i += 1;
            }
            if(i < A.length) {
                j += 1;
                A[j] = A[i];
            }
        }
        return j+1;
    }
    public static void test(){
        RemoveDuplicatesFromSortedArray instance = new RemoveDuplicatesFromSortedArray();
        int[] A = new int[]{1,1,2,3,3,3,3,3,3,4,4,5,6,6,6};
        int res = instance.removeDuplicates(A);
        System.out.println(Arrays.toString(A));
        System.out.println(res);
    }
}

File: cpp/remove_duplicates_from_sorted_array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        if(n <= 1) return n;
        int i = 0;
        int j = 1;
        while(j < n){
            if(nums[i] != nums[j]){
                nums[i+1] = nums[j];
                i += 1;
            }
            j += 1;
        }
        return i+1;
    }
};
